wallet/src/Pos/Wallet/Web/Account.hs

```diff
@@ -132,14 +132,13 @@ generateUnique desc RandomSeed generator isDuplicate = loop (100 :: Int)
                       \you are approaching the limit") desc
     loop i = do
         rand  <- liftIO randomIO
-        value <- generator rand
-        bad   <- orM
-            [ isDuplicate rand value
-            , pure $ isHardened rand  -- using hardened keys only for now
-            ]
-        if bad
-            then loop (i - 1)
-            else return value
+        if isHardened rand then
```

Comment:
Shouldn't this be `if isHardened rand then ... else loop (i - 1)` ? This change afaik was merged previously here: https://github.com/input-output-hk/cardano-sl/pull/1991/commits/4ebc883d4f08ffa13d3bce74e71792a3a54aeb42.

Reply:
@et4te as I can see we considered index which `>= firstHardened` as bad and called loop again,  but your commit considers indexes `>= firstHardened` as good. afaiu it was done intentionally, but why?

Reply:
@pva701 maybe I'm missing something, but requirements dictates we need to generate only hardened keys for addresses, for security reasons. See also here for a comment from [Dmitry](https://github.com/input-output-hk/cardano-sl/pull/1991#issuecomment-347971486), explicitly stating that what Edward fixed was a genuine bug.

Reply:
But unless I'm missing something, merging this as is would undo the bugfix in #1991 -- surely we do not want that?

I agree with @et4te that we should preserve the fix of 4ebc883, at least before deploying this. @adinapoli-iohk, you did benchmark the improvements without the bugfix already, so we do know that we have a performance improvement, and need not worry about it being masked, right?

Reply:
I'd like this to preserve the bugfix of #1991

Reply:
> so we do know that we have a performance improvement, and need not worry about it being masked, right?

Absolutely correct, Philipp, that's why I'm happy now 😉